libraries/drivers_dsl_notifies.rb

```diff
@@ -31,6 +31,18 @@ def notifies
         parsed_notifier
       end
       # rubocop:enable Metrics/LineLength
+
+      def notifying_resource(what, name, action = :restart, timing = :delayed, &block)
+        r = context.send(what, name, &block)
+        r.notifies(action, "service[#{service_name}]", timing)
+        r
+      end
+
+      %i[execute file link package template].each do |what|
+        define_method "notifying_#{what}" do |name, action = :restart, timing = :delayed, &block|
+          notifying_resource(what, name, action, timing, &block)
+        end
+      end
```

Comment:
```ruby
alias notifying_execute notifying_resource
alias notifying_file notifying_resource
alias notifying_link notifying_resource
alias notifying_package notifying_resource
alias notifying_template notifying_resource
```
won't work here? This construction is really hard to read, and you won't see the intention.
Instead of what, you can then use `__callee__` to get the method name (or `__callee__.sub('notifying_', '')`).

Reply:
I also find it hard to discern the intention in the `__callee__` construct, but your concern is valid so for complete readability please see my amended version [here](https://github.com/ajgon/opsworks_ruby/pull/115/commits/d74a56ce1ac7321e25b4fd8aa69a120115932564).